code/modules/mob/living/living_defense.dm

```diff
@@ -232,10 +232,10 @@
 /mob/living/effect_smoke(obj/effect/particle_effect/smoke/S)
 	. = ..()
 	if(!.)
-		if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO))
+		if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO) && ishuman(src))
 			smokecloak_off()
 		return
-	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO))
+	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_CAMO) && ishuman(src))
```

Comment:
Balance wise turned out to be fine, fix this

Reply:
moved to human/effect_smoke